### PR TITLE
Add eagle mass and metal mass evolution

### PR DESCRIPTION
**Pull request by MatthieuSchaller**

### CODE REVIEW

**Comment by JBorrow** [comment]
Would you be able to convert this to be consistent with the new multi-z format? https://velociraptor-python.readthedocs.io/en/latest/observational_data/creating_data.html

**Comment by JBorrow** [comment]
I think all you'd need to do is associate the lower and upper `z` with your data.

**Comment by MatthieuSchaller** [comment]
They are not really multi-z data though? more a function of z.

**Comment by MatthieuSchaller** [comment]
Ok, I have put in a wide enough range so that the points always appear.

**Comment by MatthieuSchaller** [comment]
travis reports a whole bunch of errors not related to my files.

**Comment by MatthieuSchaller** [comment]
Rebased to see whether it helps...

**Comment by MatthieuSchaller** [comment]
Re-basing saved the day. All hail our git overlords!

File: data/MetalMassDensity/conversion/convertSchaye2015_gas.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# Write everything
+processed = ObservationalData()
+processed.associate_x(a, scatter=None, comoving=True, description="Scale-factor")
+processed.associate_y(
+    rho_Z_gas, scatter=None, comoving=True, description="Gas Phase Metal Mass Density"
+)
+processed.associate_citation(citation, bibcode)
+processed.associate_name(name)
+processed.associate_comment(comment)
+processed.associate_redshift(redshift)
```
Comment (JBorrow) [inline data/MetalMassDensity/conversion/convertSchaye2015_gas.py:56]:
```suggestion
processed.associate_redshift(redshift, min(z), max(z))
```

File: data/MetalMassDensity/conversion/convertSchaye2015_star.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+processed.associate_y(
+    rho_Z_star,
+    scatter=None,
+    comoving=True,
+    description="Metal Mass locked in stars Density",
+)
+processed.associate_citation(citation, bibcode)
+processed.associate_name(name)
+processed.associate_comment(comment)
+processed.associate_redshift(redshift)
```
Comment (JBorrow) [inline data/MetalMassDensity/conversion/convertSchaye2015_star.py:59]:
```suggestion
processed.associate_redshift(redshift, min(z), max(z))
```

**Comment by MatthieuSchaller** [comment]
Done!

**Pull request merged**